src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Email/EmailAddressAttribute.cs

```diff
@@ -0,0 +1,31 @@
+using System.ComponentModel.DataAnnotations;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace OrchardCore.Email
+{
+    /// <summary>
+    /// Validates an email address.
+    /// </summary>
+    public class EmailAddressAttribute : ValidationAttribute
+    {
+        /// <inheritdoc/>
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            var emailAddressValidator = validationContext.GetService<IEmailAddressValidator>();
+
+            if (value == null)
+            {
+                return ValidationResult.Success;
```

Comment:
Nope, this similar to normal `EmailAddressAttribute` behavior, so we don't wanna confuse others

Reply:
Here the method returns a ModelState to the controller so if I would pass a NULL email it would let it pass as a valid email. The only reason this line will never get hit is that we have this condition in the controller :  

```C#
            if (String.IsNullOrEmpty(model.Email))
            {
                ModelState.AddModelError("Email", S["Email is required."]);
            }
``` 

So, if the intent is to return if the email IsValid here then it fails to do so. Because a NULL email is not an email. So the ModelState returned here I think is wrong.

Reply:
I will let others review this one but I think it's wrong unless it is by design.

Reply:
You mean this : 

https://github.com/microsoft/referencesource/blob/5697c29004a34d80acdaf5742d7e699022c64ecd/System.ComponentModel.DataAnnotations/DataAnnotations/EmailAddressAttribute.cs#L23

Then I agree

Reply:
Exactly, that's why I did the condition explicitly in the control. I know it seems weird but it's by design